contract/models/contract.py

```diff
@@ -222,72 +221,103 @@ def _prepare_invoice_line(self, line, invoice_id):
 
     @api.multi
     def _prepare_invoice(self):
-        self.ensure_one()
-        if not self.partner_id:
+        """Prepare the values for the invoice creation from the contract(s)
+        given. It's possible to provide several contracts. Only one invoice
+        will be created and most of the values will be taken from first
+        contract, but there are certain values that can be obtained from all
+        of them (for example, the origin field).
+
+        :param: self: Recordset of contract(s).
```

Comment:
It's `:param self:`.

http://www.sphinx-doc.org/en/stable/domains.html?highlight=param#python-signatures